assets/js/admin/meta-boxes-product.js

```diff
@@ -232,8 +232,14 @@ jQuery( function( $ ) {
 	$( 'input#_manage_stock' ).change( function() {
 		if ( $( this ).is( ':checked' ) ) {
 			$( 'div.stock_fields' ).show();
+			$( 'p.stock_status_field' ).hide();
 		} else {
+			var product_type = $( 'select#product-type' ).val();
+
 			$( 'div.stock_fields' ).hide();
+			if ( ! $( 'p.stock_status_field' ).hasClass( 'hide_if_' + product_type ) ) {
+				$( 'p.stock_status_field' ).show();
```

Comment:
This can be simplified to a one line selector using `:not( '.hide_if_' + product_type )`